docs/source/cheat_sheet.rst

```diff
@@ -113,6 +113,14 @@ Functions
        # type: (...) -> bool
        <code>
 
+   # Decorator functions can be expressed via generics
```

Comment:
I'd argue that this is too advanced to be included in the cheat sheet. Some of the existing things probably go a bit too deep, to be honest. My reasoning is that we don't even explain type variables in the cheat sheet, so this could be pretty confusing by itself.

Reply:
From my personal experience, I don't think we should worry about explaining anything at all here. Often the first time I'm faced with a problem, I just want a copy-paste-able example that I can bang together and see if it works. That seems exactly the point of the cheat sheet.

The fact that it's nontrivial is more reason to include it IMO.

Reply:
Okay, that's fair. I'd prefer to move this to the end of the cheat sheet, since it's more specialized than the rest of the things in this section. I like to have a narrative, where we go from basic things to less common or more advanced things.

Reply:
Sure, that makes sense. I inserted it before "Coroutines and asyncio" because it seemed less advanced than that. But maybe that's just me, I'm open to moving it to the end. :)

Reply:
Actually, I reconsidered and moved it to the end.